addon/components/connect.js

```diff
@@ -52,6 +54,44 @@ export default (stateToComputed=() => ({}), dispatchToActions=() => ({})) => {
         });
       },
 
+      /**
+       * Cache an object of attrs passed to this Component.
+       *
+       * `Component.attrs` is an object that can look like this:
+       *
+       *   {
+       *     myAttr: {
+       *       value: 'myValue'
+       *     }
+       *   }
+       *
+       * Ember provides that a `get` will return the value:
+       *
+       *   this.get('myAttr') === 'myValue'
+       *
+       * Since the user in `stateToComputed` doesn't have access to `this.get`,
+       * we take care of it in advance.  This way the user doesn't have to
+       * guess about the shape of attrs.
+       *
+       * The result is cached as `this._cachedAttrs` to avoid recomputing.
+       *
+       *   this.computeAndCacheAttrs();
+       *   console.log(this._cachedAttrs);  // Object { myAttr: 'myValue' }
+       *
+       * @method computeAndCacheAttrs
+       * @private
+       */
+      computeAndCacheAttrs() {
+        this._cachedAttrs = Object.keys(this.attrs || {}).reduce((attrs, key) => {
+          return Object.assign(attrs, { [key]: this.get(key) });
```

Comment:
@brettburley great point! i wonder if we even need to cache the result at that point:

e.g. if we do

```js
stateToComputed(redux.getState(), this.getProperties(Object.keys(this.attrs || {})));
```